sammba/registration/struct.py

```diff
@@ -435,69 +441,71 @@ def anats_to_common(anat_filenames, write_dir, brain_volume,
     if nonlinear_minimal_patches is None:
        nonlinear_minimal_patches = []
     levels_minpatches = nonlinear_levels + nonlinear_minimal_patches
-
+    out_tstat_warp_head = out_tstat_allineated_head # prevent later elif failure
+    
     for n_iter, level_or_minpatch in enumerate(levels_minpatches):
         
-        if n_iter == 0:
-            previous_warp_files = affine_transform_files
-        else:
-            previous_warp_files = warp_files
+        if n_iter == 0: previous_warp_files = affine_transform_files
         warped_files = []
         warp_files = []
-
-        out_file = fname_presuffix(centered_head_file, suffix='_warped{}'.format(n_iter))
     
         for warp_file, centered_head_file in zip(previous_warp_files, 
                                                  centered_head_files):
-
+            
+            out_file = fname_presuffix(centered_head_file,
+                                       suffix='_warped{}'.format(n_iter))
+    
             if n_iter == 0:
+                out_nwarp_cat = nwarp_cat(
+                    in_files=[('IDENT', centered_head_file), warp_file],
+                    out_file='iniwarp.nii.gz')
                 out_qwarp = qwarp(
                     in_file=centered_head_file,
                     base_file=out_tstat_allineated_head.outputs.out_file,
                     noneg=True,
                     iwarp=True,
-                    weight=nonlinear_weight,
-                    iniwarp=warp_file,
+                    weight=nonlinear_weight_file,
+                    iniwarp=[out_nwarp_cat.outputs.out_file],
                     inilev=0,
-                    maxlev=levels_minpatches[n_iter],
+                    maxlev=level_or_minpatch,
                     out_file=out_file)
                 
-            if n_iter != 0 and n_iter < len(nonlinear_levels):
+            elif n_iter < len(nonlinear_levels):
                 out_qwarp = qwarp(
                     in_file=centered_head_file,
                     base_file=out_tstat_warp_head.outputs.out_file,
                     noneg=True,
                     iwarp=True,
-                    weight=nonlinear_weight,
-                    iniwarp=warp_file,
+                    weight=nonlinear_weight_file,
+                    iniwarp=[warp_file],
                     inilev=levels_minpatches[n_iter - 1] + 1,
-                    maxlev=levels_minpatches[n_iter],
+                    maxlev=level_or_minpatch,
                     out_file=out_file)
                     
-            if n_iter != 0 and n_iter >= len(nonlinear_levels):
+            elif n_iter >= len(nonlinear_levels):
```

Comment:
I think this one should be `else`, because it is executed only when the `if` and `elif` conditions fail, so `n_iter` is systematically greater or equal to `len(nonlinear_levels)`

Reply:
done